controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentJobs.java

```diff
@@ -143,8 +143,8 @@ public boolean isDeployed(Change change) {
     /** Returns whether job has completed successfully */
     public boolean isSuccessful(Change change, JobType jobType) {
         return Optional.ofNullable(jobStatus().get(jobType))
-                .filter(JobStatus::isSuccess)
-                .filter(status -> status.lastCompletedFor(change))
+                .flatMap(JobStatus::lastSuccess)
+                .filter(status -> status.lastCompletedWas(change))
```

Comment:
This method does not affect the retrying of the job whose lastSuccess (or lastCompleted) is examined, it only determines if we can keep on rolling the Change out. 
For this purpose, it is needlessly strict. 
The proposed implementation is also in line with the javadoc, which I think is sound.

Reply:
Right, it's used to verify that e.g. test jobs passed successfully for a given change. It's best to be strict in that case. If they for some reason run a again and fail, that's wrong too and should be fixed elsewhere.

Reply:
No, it's not better to be needlessly strict: by this logic, _all_ jobs following the one which has now failed, after a previous success, should be deemed illegal, and cancelled. I don't see that happening. Also, you can't argue a method shouldn't be fixed, because there may be errors elsewhere.

Reply:
A workaround, to provide this sort of behaviour for production jobs with version upgrades, has even been committed already: `DeploymentTrigger.productionJobHasSucceededFor:444`. 
Are there reasons this isn't generally safe?